unix/x0vncserver/XDesktop.cxx

```diff
@@ -331,6 +355,217 @@ void XDesktop::keyEvent(rdr::U32 keysym, rdr::U32 xtcode, bool down) {
 void XDesktop::clientCutText(const char* str, int len) {
 }
 
+ScreenSet XDesktop::computeScreenLayout()
+{
+  ScreenSet layout;
+
+#ifdef HAVE_XRANDR
+  XRRScreenResources *res = XRRGetScreenResources(dpy, DefaultRootWindow(dpy));
+  if (!res) {
+    vlog.error("XRRGetScreenResources failed");
+    return layout;
+  }
+  vncSetGlueContext(dpy, res);
+
+  layout = ::computeScreenLayout(DefaultScreen(dpy), &outputIdMap,
+				 pb->width(), pb->height());
+  XRRFreeScreenResources(res);
+#endif
+
+  return layout;
```

Comment:
Is it really content with an empty layout when you don't have RANDR?

Reply:
Yes.